website/blog/2023-05-04-generating-dynamic-docs.md

```diff
@@ -0,0 +1,368 @@
+    - The `-n` option for `echo` removes the trailing newline
+
+The output (⚠️ except the last two characters `,%` ) can then be copy-pasted into the following:
+
+```
+dbt run-operation generate_model_yaml --args '{"model_names": [ReplaceWithYourOutputFromPreviousCommand]}'
+```
+
+Which in turn can be copy-pasted into a new `.yml` file. In our example, we write it to `_activity_based_interest.yml`.
+
+## Create docs blocks for the new columns
+
+[Docs blocks](https://docs.getdbt.com/docs/collaborate/documentation#using-docs-blocks) can be utilized to write more DRY and robust documentation. To use docs blocks normally, we could manually update our folder structure, the `.md` file, and the `.yml` file. Your file structure would look a bit like this:
+
+```
+models/core/activity_based_interest
+├── _activity_based_interest_docs.md --New docs block markdown file
+├── _activity_based_interest_docs.yml
+├── events
+│   ├── activity_based_interest_activated.sql
+│   ├── activity_based_interest_deactivated.sql
+│   ├── activity_based_interest_updated.sql
+│   ├── downgrade_interest_level_for_user.sql
+│   ├── set_inactive_interest_rate_after_july_1st_in_bec_for_user.sql
+│   ├── set_inactive_interest_rate_from_july_1st_in_bec_for_user.sql
+│   └── set_interest_levels_from_june_1st_in_bec_for_user.sql
+└── models
+    └── f_activity_based_interest.sql
+```
+
+```
+$ cat models/core/activity_based_interest/_activity_based_interest_docs.md
+{% docs activity_based_interest__id %}  
+
+Primary key of the table. See sql for key definition.
+
+{% enddocs %}
+
+{% docs activity_based_interest__user_id %}  
+
+The internal company id for a given user.
+
+{% enddocs %}
+```
+
+```
+$ cat models/core/activity_based_interest/_activity_based_interest_docs.yml
+version: 2
+
+models:
+  - name: activity_based_interest_activated
+    description: ""
+    columns:
+      - name: id
+        description: "{{ doc('activity_based_interest__id') }}"
+
+      - name: user_id
+        description: "{{ doc('activity_based_interest__user_id') }}"
+
+... (truncated for example purposes)
+
+  - name: set_inactive_interest_rate_after_july_1st_in_bec_for_user
+    description: ""
+    columns:
+      - name: id
+        description: "{{ doc('activity_based_interest__id') }}"
+
+      - name: user_id
+        description: "{{ doc('activity_based_interest__user_id') }}"
+```
+
+To confirm the formatting works, we run the following command to get dbt Docs up and running:
+
+```
+$ dbt docs && dbt docs serve
+```
+<Lightbox src="/img/blog/2023-05-04-generating-dynamic-docs/2.jpg" title="dbt Docs UI" />
+
+Here, we can confirm that the column descriptions using the doc blocks are correct.
+ 
+:::note
+Note that we would ideally like to use the `user_id` column definition from our `user_domain`, but for the sake of example, we’ll just go with it for now.
+:::
```

Comment:
In our case, user_id is a foreign key to one of our user table, where we have a docs block for user_id. So we'd want to reuse that docs block. But I reckon this might be more confusing that really improve understanding, so perhaps just scrap that whole sentence. 
```suggestion
```

Reply:
Also just realized this is already covered [here](https://github.com/dbt-labs/docs.getdbt.com/pull/3338/files/f987c1e3e96ed15cc406d3d9d0e6dc621a08568f#diff-26655aab68fa333d27ea3acf6bf69c6c3b1a4aea2f32ad67bac790d70a2ce353R338), so this note here could definitely be deleted